packages/react/src/primitives/SliderField/SliderField.tsx

```diff
@@ -90,16 +90,17 @@ const SliderFieldPrimitive: Primitive<SliderFieldProps, 'span'> = (
     [onChange]
   );
 
+  const realValue = isControlled ? value : currentValue;
   const renderedValue = React.useMemo(() => {
     const formattedValue = isFunction(formatValue)
-      ? formatValue(currentValue)
-      : currentValue;
+      ? formatValue(realValue)
+      : realValue;
     return typeof formatValue === 'string' ? (
       <View as="span">{formattedValue}</View>
     ) : (
       formattedValue
     );
-  }, [currentValue, formatValue]);
+  }, [realValue, formatValue]);
```

Comment:
Minor callout: think it probably makes sense to remove the `useMemo` here:
- if the provided `formatValue`  isn't memoized it negates the memoization here
- moving the ternary starting on line 98 inside line 141 would gate the creation of the element node when `isValueHidden` is truthy